flink-runtime/src/main/scala/org/apache/flink/runtime/jobmanager/JobManager.scala

```diff
@@ -149,6 +154,9 @@ class JobManager(
 
   var leaderSessionID: Option[UUID] = None
 
+  private var metricsRegistry : FlinkMetricRegistry = _
```

Comment:
So far, we've created the JobManager's components outside of the JobManager and passed them to it as a constructor argument. This had the advantage that if one of the components failed, it would not cause the `JobManager` to fail and the actor to die. But instead we could fail early telling exactly where the problem occurred.

Maybe we could do the same for the `FlinkMetricRegistry`.